src/js/angular/sparql-editor/controllers.js

```diff
@@ -300,6 +292,78 @@ function SparqlEditorCtrl($scope, $location, $jwtAuth, $repositories, toastr, $t
     // ================================
     // =     Setup output handlers    =
     // ================================
+
+    /**
+     * Handles the "query" event emitted by the ontotext-yasgui. The event is fired immediately before sending the
+     * request and the request object can be altered here, and it will be sent with these changes.
+     * @param {object} queryResponseEvent - the event payload containing the query and the request object.
+     */
+    function queryHandler(queryResponseEvent) {
+        updateRequestHeaders(queryResponseEvent.request);
+        changeEndpointByQueryType(queryResponseEvent.queryMode, queryResponseEvent.request);
+        const pageNumber = queryResponseEvent.request._data['pageNumber'] ? parseInt(queryResponseEvent.request._data['pageNumber']) : undefined;
+        queryResponseEvent.request._data['pageNumber'] = undefined;
+        const pageSize = queryResponseEvent.request._data['pageSize'] ? parseInt(queryResponseEvent.request._data['pageSize']) : undefined;
+        queryResponseEvent.request._data['pageSize'] = undefined;
+        if (pageSize && pageNumber) {
+            queryResponseEvent.request._data['offset'] = (pageNumber - 1) * (pageSize -1);
+            queryResponseEvent.request._data['limit'] = pageSize;
+        }
+    }
+    outputHandlers.set(EventDataType.QUERY, queryHandler);
+
+    /**
+     * Handles the "countQuery" event emitted by the ontotext-yasgui. The event is fired immediately before sending the
+     * count query request and the request object can be altered here, and it will be sent with these changes.
+     * @param {object} countQueryResponseEvent - the event payload containing the query and the request object.
+     */
+    function countQueryResponseHandler(countQueryResponseEvent) {
+        updateRequestHeaders(countQueryResponseEvent.request);
+        changeEndpointByQueryType(countQueryResponseEvent.queryMode, countQueryResponseEvent.request);
+        countQueryResponseEvent.request._data['pageSize'] = undefined;
+        countQueryResponseEvent.request._data['pageNumber'] = undefined;
+        countQueryResponseEvent.request._data['count'] = 1;
+    }
+    outputHandlers.set(EventDataType.COUNT_QUERY, countQueryResponseHandler);
+
+    function extractTotalElements(countResponse) {
+        if (!countResponse || !countResponse.body) {
```

Comment:
Here as well we can benefit from a response mapper and a model the parsing and logic can be encapsulated